platform/solr/solr-factory/src/main/java/org/codice/solr/factory/impl/HttpSolrClientFactory.java

```diff
@@ -0,0 +1,305 @@
+        }
+        return client;
+    }
+
+    private static CloseableHttpClient getHttpClient(boolean retryRequestsOnError) {
+        SSLConnectionSocketFactory sslConnectionSocketFactory = new SSLConnectionSocketFactory(
+                getSslContext(), getProtocols(), getCipherSuites(),
+                SSLConnectionSocketFactory.BROWSER_COMPATIBLE_HOSTNAME_VERIFIER);
+        HttpRequestRetryHandler solrRetryHandler = new SolrHttpRequestRetryHandler();
+
+        HttpClientBuilder builder = HttpClients.custom()
+                .setSSLSocketFactory(sslConnectionSocketFactory)
+                .setDefaultCookieStore(new BasicCookieStore())
+                .setMaxConnTotal(128)
+                .setMaxConnPerRoute(32);
+
+        if (retryRequestsOnError) {
+            builder.setRetryHandler(solrRetryHandler);
+        }
+
+        return builder.build();
+    }
+
+    private static String[] getProtocols() {
+        if (System.getProperty("https.protocols") != null) {
+            return StringUtils.split(System.getProperty("https.protocols"), ",");
+        } else {
+            return DEFAULT_PROTOCOLS.toArray(new String[DEFAULT_PROTOCOLS.size()]);
+        }
+    }
+
+    private static String[] getCipherSuites() {
+        if (System.getProperty("https.cipherSuites") != null) {
+            return StringUtils.split(System.getProperty("https.cipherSuites"), ",");
+        } else {
+            return DEFAULT_CIPHER_SUITES.toArray(new String[DEFAULT_CIPHER_SUITES.size()]);
+        }
+    }
+
+    private static SSLContext getSslContext() {
+        if (System.getProperty("javax.net.ssl.keyStore") == null ||
+                System.getProperty("javax.net.ssl.keyStorePassword") == null ||
+                System.getProperty("javax.net.ssl.trustStore") == null ||
+                System.getProperty("javax.net.ssl.trustStorePassword") == null) {
+            throw new IllegalArgumentException(
+                    "KeyStore and TrustStore system properties must be set.");
+        }
+
+        KeyStore trustStore = getKeyStore(System.getProperty("javax.net.ssl.trustStore"),
+                System.getProperty("javax.net.ssl.trustStorePassword"));
+        KeyStore keyStore = getKeyStore(System.getProperty("javax.net.ssl.keyStore"),
+                System.getProperty("javax.net.ssl.keyStorePassword"));
+
+        SSLContext sslContext = null;
+
+        try {
+            sslContext = SSLContexts.custom()
+                    .loadKeyMaterial(keyStore, System.getProperty("javax.net.ssl.keyStorePassword")
+                            .toCharArray())
+                    .loadTrustMaterial(trustStore)
+                    .useTLS()
+                    .build();
+        } catch (UnrecoverableKeyException | NoSuchAlgorithmException | KeyStoreException |
+                KeyManagementException e) {
+            LOGGER.warn("Unable to create secure HttpClient. The Solr Client should not be used in this state.", e);
```

Comment:
`SSLConnectionSocketFactory` makes sure SSLContext is not null but I will explicitly throw an exception in case we switch to another SSL socket factory that does not have that check.